### ISSUE TITLE
Absolute path in command line containing spaces

### ISSUE THREAD

**Issue by kevinboulain** [systemd/systemd#2132]
Hi there,

Today, I needed to quote a path that may contain spaces but I could not find the correct way to write the _.service_ file.
I tried several things, here is the minimal setup:

```
# cat > /tmp/space\ bin << EOF
#!/bin/sh
echo blah > /tmp/log
EOF
# chmod +x  /tmp/space\ bin

# cat > /etc/systemd/system/path-with-spaces.service << EOF
[Unit]
Description=path with spaces
After=syslog.service

[Service]
ExecStart='/tmp/space bin'
Type=oneshot

[Install]
WantedBy=multi-user.target
EOF
# systemctl daemon-reload
# systemctl status -l path-with-spaces
● path-with-spaces.service - path with spaces
   Loaded: error (Reason: Invalid argument)
   Active: inactive (dead)

Dec 10 09:47:45 host systemd[1]: [/etc/systemd/system/path-with-spaces.service:6] Executable path is not absolute, ignoring: '/tmp/space bin'
Dec 10 09:47:45 host systemd[1]: path-with-spaces.service lacks ExecStart setting. Refusing.
Dec 10 09:48:10 host systemd[1]: [/etc/systemd/system/path-with-spaces.service:6] Executable path is not absolute, ignoring: '/tmp/space bin'
Dec 10 09:48:10 host systemd[1]: path-with-spaces.service lacks ExecStart setting. Refusing.
```

With a quick search, it seems this is what is causing the error message:
https://github.com/systemd/systemd/blob/564c44436cf64adc7a9eff8c17f386899194a893/src/core/load-fragment.c#L635
And this is the the function that test for an absolute path:
https://github.com/systemd/systemd/blob/375f1b0491568fc4e31fee95364d7c855371167e/src/basic/path-util.c#L48

So quotes are not supported for the executable path yet?

Backslash (_\_) does not allow to escape the space in the path too (but it is not indicated in the docs anyway).

Maybe related to #624?

On a side note, is it really necessary to try to reload the _.service_ file even if it contain a syntax error (see pasted log above)?

Regards.

EDIT:
By the way, I'm on a Debian Jessie system (if the information is useful):

```
# systemd --version
systemd 215 +PAM +AUDIT +SELINUX +IMA +SYSVINIT +LIBCRYPTSETUP +GCRYPT +ACL +XZ -SECCOMP -APPARMOR
```

**Comment by n8henrie** [comment]
I think I have the same issue on Raspbian Jessie, can't seem to use a service file with a space in the path.

[Thought it would work](http://www.freedesktop.org/software/systemd/man/systemd.service.html#Command%20lines):

> The command to execute must be an absolute path name. It may contain spaces, but control characters are not allowed.

I've tried `ExecStart=`
- `/path/with spaces/executable` -> `Failed at step EXEC spawning /path/with: No such file or directory` 
- `"/path/with spaces/executable"` -> `Not an absolute path, ignoring:` 
- `systemd-escape --path "/path/with spaces/executable"` -> `path-with\x20spaces-executable"` -> `Not an absolute path, ignoring:`
- `/path/with\sspaces/executable` -> `Failed at step CHDIR spawning /path/with spaces/executable: No such file or directory`
- `/path/with\ spaces/executable` -> `The process /path/with\ spaces/executable could not be executed and failed.
- `/path/"with spaces"/executable` -> `Failed at step EXEC spawning /path/"with: No such file or directory`
- `/path/without_spaces/executable` -> Runs without issue

```
$ systemd --version
systemd 215
+PAM +AUDIT +SELINUX +IMA +SYSVINIT +LIBCRYPTSETUP +GCRYPT +ACL +XZ -SECCOMP -APPARMOR
```

EDIT: Found an acceptable workaround: `ExecStart=/usr/bin/env "/path/with spaces/executable"`

**Comment by evverx** [comment]
Fixed: c853953658865ec3b

**Comment by chrisgoldberg1** [comment]
Not really fixed.

```
systemd 257 (257.8-2-arch)
+PAM +AUDIT -SELINUX -APPARMOR -IMA +IPE +SMACK +SECCOMP +GCRYPT +GNUTLS +OPENSSL +ACL +BLKID +CURL +ELFUTILS +FIDO2 +IDN2 -IDN +IPTC +KMOD +LIBCRYPTSETUP +LIBCRYPTSETUP_PLUGINS +LIBFDISK +PCRE2 +PWQUALITY +P11KIT +QRENCODE +TPM2 +BZIP2 +LZ4 +XZ +ZLIB +ZSTD +BPF_FRAMEWORK +BTF +XKBCOMMON +UTMP -SYSVINIT +LIBARCHIVE
```

Same issue. only `/usr/bin/env` workaround is working without specifying WorkDir.
`WorkingDirectory= path is not absolute:`
